File: Lesson8 Algorithms/Problem 1/Solutions/Solution.cpp
#include <iostream>
#include<cmath>

using namespace std;

bool isPrime(int n){
    if (n <= 1) return false;
    if (n == 2) return true;
    for(int i = 2; i <= sqrt(n); i++)
        if(n % i == 0) return false;
    return true;
}
int main()
{
   int n;
   cin>>n;
   while(true){
       if(isPrime(n)){
           cout<<n<<endl;
           break;
       }
       n++;
   }
   
   return 0;
}

File: Lesson8 Algorithms/Problem 1/Solutions/Solution.py
# python 2 code

import math
    
def isPrime(n):
    if n == 2:
        return True;
    if n <= 1:
        return False;
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
    
    return True
    
n = int(raw_input())

while True:
    if isPrime(n):
        print n
        break
    n = n + 1
